### ISSUE TITLE
error with Explain CML / imatinib

### ISSUE THREAD

**Issue by andrewsu** [biothings/biothings_explorer#27]
I am running this code snippet (ignoring for the moment that I'm assigning the object for CML to a variable called `asthma`):

```
ht = Hint()
# find all potential representations of asthma
asthma_hint = ht.query("chronic myelogenous leukemia")
# select the correct representation of asthma
asthma = asthma_hint['DiseaseOrPhenotypicFeature'][0]
imatinib_hint = ht.query("imatinib")
# select the correct representation of imatinib
imatinib = imatinib_hint['ChemicalSubstance'][0]
fc = FindConnection(input_obj=asthma, output_obj=imatinib, intermediate_nodes='Gene')
fc.connect(verbose=True)
```
All seems to execute well -- log ends with this:

```
BTE found 51 unique intermediate nodes connecting 'chronic myelogenous leukemia' and 'IMATINIB'
```

But I get an error when I try to print the results:
```
df = fc.display_table_view()
df
```
Error message:
```
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-48-78b9fddae8b8> in <module>
----> 1 df = fc.display_table_view()
      2 df

~/env/bte_schema/lib/python3.6/site-packages/biothings_explorer/user_query_dispatcher.py in display_table_view(self)
    766         >>> df
    767         """
--> 768         return self.fc.display_table_view()

~/env/bte_schema/lib/python3.6/site-packages/biothings_explorer/user_query_dispatcher.py in display_table_view(self)
    475         """
    476         paths = self.show_path()
--> 477         return connect_networkx_to_pandas_df(self.G, paths)
    478 
    479 

~/env/bte_schema/lib/python3.6/site-packages/biothings_explorer/networkx_helper.py in connect_networkx_to_pandas_df(G, paths, pred1, intermediate, intermediate_type, pred2)
    181                              'pred2_source': retrieve_prop_from_edge(v, 'source'),
    182                              'pred2_api': retrieve_prop_from_edge(v, 'api'),
--> 183                              'pred2_pubmed': retrieve_prop_from_edge(v, 'pubmed'),
    184                              'output': _path[2],
    185                              'output_type': G.nodes[_path[2]]['type']})

~/env/bte_schema/lib/python3.6/site-packages/biothings_explorer/networkx_helper.py in retrieve_prop_from_edge(edge_info, prop)
    154             if type(data) != list:
    155                 data = [data]
--> 156             return ','.join(data)
    157 
    158 

TypeError: sequence item 0: expected str instance, int found
```

**Comment by andrewsu** [comment]
Same error if I replace the CML object with the following:
```
asthma_hint = ht.query("Acute Lymphoblastic Leukemia")
# select the correct representation of asthma
asthma = asthma_hint['DiseaseOrPhenotypicFeature'][1]
```

**Comment by kevinxin90** [comment]
Fixed in: 1ba749a7ef99671d361355cc2cb028039de5dce7

**Comment by andrewsu** [comment]
confirmed that this issue has been fixed...

**Comment by andrewsu** [comment]
Hmm, reopening this issue because I think I *might* have noticed a related issue (but LMK if it should be a new ticket).

When running this code:

```
ht = Hint()
# find all potential representations of hyperphenylalaninemia
hyperphenylalaninemia_hint = ht.query("TP53")
# select the correct representation of hyperphenylalaninemia
hyperphenylalaninemia = hyperphenylalaninemia_hint['Gene'][0]
fc = FindConnection(input_obj=hyperphenylalaninemia, output_obj='ChemicalSubstance', intermediate_nodes=None)
fc.connect(verbose=True)
df = fc.display_table_view()
```

I get this error:
```
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-8-4704f2c0fe62> in <module>
----> 1 df = fc.display_table_view()
      2 

~/env/bte_schema/lib/python3.6/site-packages/biothings_explorer/user_query_dispatcher.py in display_table_view(self)
    766         >>> df
    767         """
--> 768         return self.fc.display_table_view()

~/env/bte_schema/lib/python3.6/site-packages/biothings_explorer/user_query_dispatcher.py in display_table_view(self)
    675         """
    676         paths = self.show_path()
--> 677         return connect_networkx_to_pandas_df(self.G, paths)
    678 
    679 class FindConnection:

~/env/bte_schema/lib/python3.6/site-packages/biothings_explorer/networkx_helper.py in connect_networkx_to_pandas_df(G, paths, pred1, intermediate, intermediate_type, pred2)
    165     data = []
    166     for _path in paths:
--> 167         output_id = get_primary_id_from_equivalent_ids(G.nodes[_path[-1]].get('equivalent_ids'), G.nodes[_path[-1]]['type'])
    168         output_name = get_name_from_equivalent_ids(G.nodes[_path[-1]].get('equivalent_ids'))
    169         if len(_path) == 3:

~/env/bte_schema/lib/python3.6/site-packages/biothings_explorer/utils.py in get_primary_id_from_equivalent_ids(equivalent_ids, _type)
     40     for _item in id_rank:
     41         if equivalent_ids.get(_item):
---> 42             return (_item[4:] + ':' + equivalent_ids[_item][0])
     43     # if no id found, return a random one from equivalent ids
     44     for k, v in equivalent_ids.items():

TypeError: must be str, not int
```

**Comment by kevinxin90** [comment]
Fixed in 0575778d111b347df56622484dfe5e2a186b6928

**Comment by andrewsu** [comment]
confirmed the fix!
